provider/resource_keycloak_users_permissions_test.go

```diff
@@ -0,0 +1,253 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/mrparkers/terraform-provider-keycloak/keycloak"
+)
+
+func TestAccKeycloakUsersPermission_basic(t *testing.T) {
+	realmName := "tf_view-" + acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: testAccProviderFactories,
+		PreCheck:          func() { testAccPreCheck(t) },
+		Steps: []resource.TestStep{
+			{
+				Config: testKeycloakUsersPermission_basic(realmName),
+				Check:  testAccCheckKeycloakUsersPermissionExists("keycloak_users_permissions.my_permission"),
+			}, {
+				ResourceName:      "keycloak_users_permissions.my_permission",
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateId:     realmName,
+			}, {
+				Config: testKeycloakUsersPermissionDelete_basic(realmName),
+				Check:  testAccCheckKeycloakUsersPermissionDoesNotExists(realmName),
+			},
+		},
+	})
+}
+
+func testAccCheckKeycloakUsersPermissionExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		permissions, err := getUsersPermissionsFromState(s, resourceName)
+		if err != nil {
+			return err
+		}
+		rs, ok := s.RootModule().Resources[resourceName]
+
+		if !ok {
+			return fmt.Errorf("resource not found: %s", resourceName)
+		}
+		authorizationResourceServerId := rs.Primary.Attributes["authorization_resource_server_id"]
+
+		var realmManagementId string
+		clients, _ := keycloakClient.GetOpenidClients(permissions.RealmId, false)
+		for _, client := range clients {
+			if client.ClientId == "realm-management" {
+				realmManagementId = client.Id
+				break
+			}
+		}
+
+		if authorizationResourceServerId != realmManagementId {
+			return fmt.Errorf("computed authorizationResourceServerId %s was not equal to %s (the id of the realm-management client)", authorizationResourceServerId, realmManagementId)
+		}
+
+		viewScopePolicyId := rs.Primary.Attributes["view_scope.0.policies.0"]
```

Comment:
@mrparkers Is there a better way to get the values ?

Reply:
Not that I know of 😞